utils/promqueries.py

```diff
@@ -145,4 +151,338 @@ def get_all_queries(timestamp_tuples, step):
+    else:
+        try:
+            df = MetricRangeDataFrame(list(results_dict.values()))
+            if deduplicate:
+                df = df[~df.index.duplicated(keep='first')]
+            results = [df]
+        except:
+            errors.append(results_dict)
+            
+    if len(results) > 0:
+        temp = results
+        return temp, errors
+    else:
+        return None, None
+
+def get_all_queries_v3(timestamp_tuples, 
+                       step="120s", 
+                       granular_step=None, 
+                       namespace='triton', 
+                       deduplicate=True,
+                       prom=PrometheusConnect(url="http://lsdataitb.fnal.gov:9009/prometheus", disable_ssl=True)
+                      ):
+    """Function for running most common prometheus queries for analyzing server behavior.
+    step: time step for the query window, will be the index of resulting pandas dataframes. Should be minimally 4x as long as the most infrequent metric gathered, which at the EAF is 30s x 4 = 120s
+    granular_step: parameter for auto-discovery of unique models and GPU instances in the active timeframe. Should subdivide the timestampe range into a minimum of subdivisions, i.e. use "1d" for a total range of "8d"
+    namespace: Denotes which namespace to use for the metrics. At the LPC EAF, should be 'triton'
+    deduplicate: Removes duplicate timestamps, which may be introduced with overlapping timestamp_tuples starts and stops"""
+    
+    rs = ""
+    rsm = ""
+    if isinstance(namespace, str):
+        rs = "{namespace='"+namespace+"'}"
+        rsm = ",namespace='"+namespace+"'"
+    # A dictionary for our results
+    results = {}
+    errors = {}
+    # Tuples of the queries we'll make, for debugging and info
+    queries = []
+    
+    # Some queries are best created after understanding which unique models+version have been run in the triton servers
+    # and which GPU instances have been active. These are then used to formulate model/version-specific and GPU-specific stats
+    columns_step = granular_step if granular_step is not None else step
+    unique_model_versions, inactive_model_versions = find_active_models(timestamp_tuples, step=columns_step, namespace=namespace, prom=prom)
+    unique_gpu_instances = find_active_gpus(timestamp_tuples, step=columns_step, namespace=None, prom=prom) #different namespace entirely
+    
+    #Basic queries. Some of them are used as proxies to figure out the unqique queries to make later, like the "gpu_tensor_util" below
+    for key, query in track({
+        "num_instances": "count((sum by(pod) (delta(nv_inference_request_success"+rs+"["+step+"]))) > 0)",
+        "inf_rate_net":"sum (rate(nv_inference_count"+rs+"["+step+"]))",
+        "inf_reqs_net":"sum(rate(nv_inference_request_success"+rs+"["+step+"]))",
+        "inf_req_dur_net": "avg (delta(nv_inference_request_duration_us"+rs+"["+step+"])/(0.001+delta(nv_inference_request_success"+rs+"["+step+"])))",
```

Comment:
```suggestion
        # These queries average over all triton instances (avg op)
        # in the time slice, take the *_duration (delta op on the time range vector) and
        # divide by the number of successful requests  in the same duration 
        # (requests != inferences or batch executions) with an additional 0.001 to avoid div-by-zero
        "inf_req_dur_net": "avg (delta(nv_inference_request_duration_us"+rs+"["+step+"])/(0.001+delta(nv_inference_request_success"+rs+"["+step+"])))",
```